modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/rel/ExecutionTest.java

```diff
@@ -55,14 +57,16 @@
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
 
 /**
  * ExecutionTest.
  * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
  */
 public class ExecutionTest extends AbstractExecutionTest<Object[]> {
-    @Test
-    public void testSimpleExecution() {
+    @ParameterizedTest(name = "join algo : {0}")
+    @ValueSource(strings = {"NLJoin", "HashJoin"})
+    public void testSimpleExecution(String joinImplementation) {
```

Comment:
please move all tests related to HashJoinNode to dedicated class

Reply:
i make some refactoring there